### PR TITLE
Drop guzzle

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this change provide BC-break.

### CODE REVIEW

File: src/Block/Social/TwitterEmbedTweetBlockService.php [hunk 1]
```diff
@@ -41,7 +40,7 @@
  *
  * @author Hugo Briand <briand@ekino.com>
  */
-class TwitterEmbedTweetBlockService extends BaseTwitterButtonBlockService implements EditableBlockService
+final class TwitterEmbedTweetBlockService extends BaseTwitterButtonBlockService implements EditableBlockService
```
Comment (VincentLanglet) [inline src/Block/Social/TwitterEmbedTweetBlockService.php:43]: We should `@final` first in the stable branch (for this class and others)

Comment (wbloszyk) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:43]: #517

File: UPGRADE-3.0.md [hunk 2]
```diff
@@ -1,9 +1,19 @@
 UPGRADE FROM 2.X to 3.0
 =======================
 
+## guzzlehttp/guzzle
+
+If you using `sonata.seo.block.twitter.embed` service with guzzle, you must create custom service based on guzzle client and add it to configuration:
```
Comment (phansys) [inline UPGRADE-3.0.md:6]:
```suggestion
If you are using `sonata.seo.block.twitter.embed` service with Guzzle, you MUST create a custom service based on the Guzzle client and add it to configuration:
```

File: UPGRADE-3.0.md [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
+
+    sontata_seo:
+        http:
+            client: `you_custom.guzzle_client` # Psr\Http\Client\ClientInterface
```
Comment (phansys) [inline UPGRADE-3.0.md:10]:
```suggestion
            client: `your_custom.guzzle_client` # Psr\Http\Client\ClientInterface
```

File: UPGRADE-3.0.md [hunk 4]
```diff
@@ -1,9 +1,19 @@
...
+            message_factory: `you_custom.message_facory` # Psr\Http\Message\RequestFactoryInterface
```
Comment (phansys) [inline UPGRADE-3.0.md:11]:
```suggestion
            message_factory: `your_custom.message_facory` # Psr\Http\Message\RequestFactoryInterface
```

File: src/Block/Social/TwitterEmbedTweetBlockService.php [hunk 5]
```diff
@@ -217,52 +216,20 @@ private function loadTweet(BlockContextInterface $blockContext): ?string
...
-            return $apiTweet['html'];
-        } catch (GuzzleException $e) {
+            $response = $this->httpClient->sendRequest(
+                $this->messageFactory->createRequest(
+                    'GET',
+                    $this->buildUri($uriMatched, $blockContext->getSettings())
+                )
+            );
+        } catch (ClientExceptionInterface $e) {
             // log error
```
Comment (phansys) [inline src/Block/Social/TwitterEmbedTweetBlockService.php:227]: This comment is a TODO?

**Review by phansys — CHANGES_REQUESTED** [review]

File: src/Block/Social/TwitterEmbedTweetBlockService.php (hunk 5 above)
Comment (wbloszyk) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:227]: It is no my comment. IMO it is TODO.

Comment (phansys) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:227]: I know, I just saw it during the review.
IMO, we should add a "TODO" or "@todo" prefix there.

Comment (wbloszyk) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:227]: This change is not in this PR scope.
I want resolve and prepare master to major release in nex week, so I will do it in other PR.

**Review by phansys — DISMISSED** [review]

**Review by VincentLanglet — DISMISSED** [review]

File: src/Block/Social/TwitterEmbedTweetBlockService.php [hunk 6]
```diff
@@ -164,7 +163,7 @@ public function getMetadata(): MetadataInterface
     /**
      * Returns supported API parameters from settings.
      */
-    protected function getSupportedApiParams(): array
+    public function getSupportedApiParams(): array
```
Comment (core23) [inline src/Block/Social/TwitterEmbedTweetBlockService.php:166]: Why this change?

File: src/Block/Social/TwitterEmbedTweetBlockService.php [hunk 7]
```diff
@@ -182,7 +181,7 @@ protected function getSupportedApiParams(): array
     /**
      * Builds the API query URI based on $settings.
      */
-    protected function buildUri(bool $uriMatched, array $settings): string
+    public function buildUri(bool $uriMatched, array $settings): string
```
Comment (core23) [inline src/Block/Social/TwitterEmbedTweetBlockService.php:184]: Same here?

**Review by core23 — CHANGES_REQUESTED** [review]

File: src/Block/Social/TwitterEmbedTweetBlockService.php (hunk 7 above)
Comment (wbloszyk) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:184]: It is a getter with string return. Make it public will not change functionality but will add extra information require in tests.

File: src/Block/Social/TwitterEmbedTweetBlockService.php (hunk 6 above)
Comment (wbloszyk) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:166]: This method always return the same array. Access to possible params for Twitter API should be public to allow validate it by own. Now unsupported params are ignored. 

BTW Mybe move it co constans will be better solution, becouse it depends on API, not object.

Comment (VincentLanglet) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:166]: A const is better.

File: src/Block/Social/TwitterEmbedTweetBlockService.php (hunk 7 above)
Comment (VincentLanglet) [reply src/Block/Social/TwitterEmbedTweetBlockService.php:184]: We should not add public just to help the tests.

It's better to not expose extra method than the interface ones.

Comment (wbloszyk) [reply src/Block/Social/TwitterEmbedTweetBlockService.php]: I revert this change and final modifier. Lets make it in other PR.

**Review by core23 — APPROVED** [review]

**Pull request merged**